lib/phase/functional.js

```diff
@@ -141,6 +142,23 @@ function generateWorkflow(doc) {
 }
 
 /**
+ * Get the defined workflowGraph, or if missing generate one (list of all jobs in series)
+ * @method generateWorkflowGraph
+ * @param  {Object}          doc Document that went through flattening
+ * @return {Object}              WorkflowGraph that consists of nodes and edges
+ */
+function generateWorkflowGraph(doc) {
+    // In case doc got changed during workflow parsing
+    const cloneDoc = clone(doc);
+
+    if (doc.workflow) {
+        return WorkflowParser.getWorkflow(cloneDoc, { useLegacy: true });
+    }
+
+    return WorkflowParser.getWorkflow(cloneDoc);
```

Comment:
This is a difficult call to make here. We have projects that do not define workflow but do have more than one job. The default behavior in legacy mode is to convert the list of jobs into a linear workflow. In non-legacy mode, this will be N detached pipelines.

I think what we want to do here is actually _always_ run workflow through legacy-mode, the parser will do the right thing if it detects non-legacy `requires` configs. When we are satisfied that we no longer wish to support legacy mode, we just stop passing in the `useLegacy` flag.

Reply:
Based on the logic we have already, in that case, it will always generate the legacy workflow.
https://github.com/screwdriver-cd/config-parser/blob/master/lib/phase/functional.js#L191

So after that when it comes to `generateWorkflowGraph`, we will have the workflow already.

Reply:
So our legacy pipelines that do not explicitly define a workflow already have something defining a workflow before it gets here? We should consider removing those bits, since workflow parser should be handling that, but I guess we can do that later.